File: MCCME/3574.py
def is_prime(n):
    d = 2
    while d * d <= n:
        if n % d == 0:
            return 0
        d += 1
    return 1


size = int(input())

res = [0 for _ in range(size + 1)]
res[0] = res[1] = 1
for i in range(2, len(res)):
    if i == 2 or not is_prime(i):
        res[i] = res[i - 1] + res[i - 2]

print(res[size])

File: LeetCode/3574.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <set>
#include <map>
#include <string>
using namespace std;

int main() {
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);

    int n, k;
    cin >> n >> k;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        int x, y;
        cin >> x >> y;

        a[i] = x * y;
    }

    int cur = 0;
    for (int i = 0; i < k; i++)
        cur += a[i];

    int res = cur;
    for (int i = k; i < n; i++) {
        cur += a[i] - a[i - k];
        res = max(res, cur);
    }

    cout << res;
}